File: 120. Triangle/120. Triangle.rb
def minimum_total(triangle)
    dp = triangle.last.dup
    
    (triangle.size - 2).downto(0) do |i|
        (0..i).each do |j|
            dp[j] = triangle[i][j] + [dp[j], dp[j + 1]].min
        end
    end
    
    dp[0]
end

File: 120. Triangle/120. Triangle.rs
impl Solution {
    pub fn minimum_total(triangle: Vec<Vec<i32>>) -> i32 {
        let mut dp = triangle.last().unwrap().clone();
        
        for i in (0..triangle.len()-1).rev() {
            for j in 0..=i {
                dp[j] = triangle[i][j] + dp[j].min(dp[j + 1]);
            }
        }
        
        dp[0]
    }
}